### PR TITLE
more GitHub Actions enablement

### PR DESCRIPTION
**Pull request by calebcartwright**
Follow up to #3969 that enables GHA for Linux, Mac, and the Integration test jobs that are currently executed on Travis. I decided to go ahead and work on this since some of the other issues I was eyeing are blocked 😄 

Similar to my experience enabling GHA for Windows in #3969, I found a few things took a little longer to set up due to the immaturity of the platform/features that haven't yet been released. However, overall it still seems really promising with several opportunities for continued improvement.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        profile: minimal
+        default: true
+    - name: build
+      run: |
+        rustc -Vv
+        cargo -V
+        cargo build
+    - name: test
+      run: cargo test
+    - name: 'test ignored'
```
Comment (calebcartwright) [inline .github/workflows/linux.yml]: I'll update this (and the others) to account for the recent change splitting into multiple crates

**Comment by calebcartwright** [comment]
I also went ahead and added a few cargo aliases as part of this to simplify the command args for running tests, and updated the GHA workflows to leverage those aliases.

For example `cargo test-all` (also aliased as `cargo t`) is now an alias that will run `cargo test --manifest-path rustfmt-core/Cargo.toml` (plus whatever other args/options provided to cargo test/libtest).

**Comment by topecongiro** [comment]
@calebcartwright Thank you so much for setting GitHub Action in this repository!

I'm all for migrating to GitHub Action. We are not worried about being locked-in to GitHub (as long as the rust itself is hosted on GitHub we must stay as well :shrug: ) and Travis is... has been slow and unstable, to say the least.

Having said that, I'm still skeptical about the stability of GitHub Action. In particular, I don't know enough about whether Rust's support is sufficient. As such, I will keep using Travis and AppVeyor along with GitHub Action for a while.

**Comment by calebcartwright** [comment]
> As such, I will keep using Travis and AppVeyor along with GitHub Action for a while.

Yup that makes sense. GHA has been running well for Windows world for a while now for us, will be interesting to see how the Linux and Mac jobs hold up on GHA as well

**Pull request merged**